### PR TITLE
monitoringdashboard: add timeSeriesTable to widget

### PR DESCRIPTION
**Pull request by justinsb**
- **monitoringdashboard: add timeSeriesTable to widget**

### CODE REVIEW

File: pkg/controller/direct/monitoring/refs.go [hunk 1]
```diff
@@ -101,5 +186,21 @@ func (r *refNormalizer) VisitField(path string, v any) error {
...
 	}
+	if alertChart, ok := v.(*krm.AlertChart); ok {
+		if ref, err := normalizeMonitoringAlertPolicyRef(r.ctx, r.kube, r.src, r.project, alertChart.AlertPolicyRef); err != nil {
+			return err
+		} else {
+			alertChart.AlertPolicyRef = ref
+		}
+	}
+
+	if projectRef, ok := v.(*refs.ProjectRef); ok {
```
Comment (justinsb) [inline pkg/controller/direct/monitoring/refs.go:197]: This is sneaky ... these are still fields!  So we could actually have `refs.ProjectRef` implement a Normalize interface, and then we have references just work automatically with no extra controller code.

File: pkg/test/resourcefixture/testdata/basic/monitoring/v1beta1/monitoringdashboard/monitoringdashboardfull/_generated_export_monitoringdashboardfull.golden [hunk 2]
```diff
@@ -85,7 +85,7 @@ spec:
       widget:
         alertChart:
           alertPolicyRef:
-            external: projects/${projectId}/alertPolicies/1719436193634641396
+            external: projects/${projectId}/alertPolicies/1719437683981846099
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/monitoring/v1beta1/monitoringdashboard/monitoringdashboardfull/_generated_export_monitoringdashboardfull.golden:88]: Looks like we need a normalize override?

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

Only reviewed the last commit for `timeSeriesTable`. It looks great. I guess the other commits are covered in other PRs (and will be reviewed separately).

File: pkg/test/resourcefixture/testdata/basic/monitoring/v1beta1/monitoringdashboard/monitoringdashboardfull/_generated_export_monitoringdashboardfull.golden (hunk 2 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/monitoring/v1beta1/monitoringdashboard/monitoringdashboardfull/_generated_export_monitoringdashboardfull.golden:88]: Yes, I think this is an earlier PR so should go away in a rebase!

**Comment by justinsb** [comment]
(Tried resolving the conflict in the GitHub web editor to see if that kept LGTM; it did not, so resolved and pushed)

**Pull request merged**